calendar-bundle/contao/classes/Calendar.php

```diff
@@ -410,7 +395,14 @@ protected function addEvent($objEvent, $intStart, $intEnd, $strUrl, $strBase='',
 				break;
 
 			default:
-				$link = sprintf(preg_replace('/%(?!s)/', '%%', $strUrl), ($objEvent->alias ?: $objEvent->id));
+				$link = $objParent->getAbsoluteUrl('/' . ($objEvent->alias ?: $objEvent->id));
+
+				// Backwards compatibility (see #8329)
+				if ($strBase && !preg_match('#^https?://#', $link))
```

Comment:
This should never be necessary if we generate the URL here ourself.

Reply:
Unrelated to this PR.
This was added in https://github.com/contao/core/pull/8329 @fritzmg do you know if this can be removed?

Reply:
How is that not related? Previously the URL was passed in to the method and could therefore be relative. Now it uses `PageModel::getAbsoluteUrl` so the http-Regex can never be true?

Reply:
> How is that not related?

This code was just moved, I didn’t change that.

> Previously the URL was passed in to the method and could therefore be relative. Now it uses PageModel::getAbsoluteUrl so the http-Regex can never be true?

Same as before, see https://github.com/contao/contao/blob/e5995ffa484b0dddb8acdc20eb751f0e976ddcd9/calendar-bundle/contao/classes/Calendar.php#L176-L180

Reply:
That check was actually added in https://github.com/contao/contao/commit/d822ca8da20bdf30656df20088ba53045a5b0548 (not in my original PR).

But in any case this is not related to this PR.

Reply:
I don't understand why this is so complicated. Yes obviously the code was just moved, but it was to sanitize a method argument that _anyone_ could have called. Now noone can call it/we know it's already same, why would we still want to sanitize?

Reply:
I agree with @aschempp. Lines 399 to 405 can be deleted as the URL will never be relative.

Reply:
As this is not related to this pull request and also affects Cotnao 4.9, I will create seperate pull requests for that issue.

Reply:
See https://github.com/contao/contao/pull/5152

Reply:
why? This PR makes the base obsolete, why would you remove the argument in a separate PR and not here directly?

Reply:
> This PR makes the base obsolete
How does it do that? I don’t understand what you mean by that.

Reply:
The check was likely there for a use-case where you extend from `Contao\Calendar` and then call

```
$this->addEvent(…)
```

with a custom URL. In this case the method wants to check whether the URL has scheme and host or not and prepend the base for the latter case - because it was previously ok to pass a non-absolute URL.

In this PR you are changing the `addEvent` method so that you are not able to pass a custom URL anymore and thus the `if ($strBase && !preg_match('#^https?://#', $strUrl))` can be omitted altogether there.

The question then is:

1. Do we want to keep the possibility to pass a custom URL to `addEvent`?
2. If not, should we may be make the method private and deprecate it in 4.13?

Reply:
I integrated this now in this PR: 54e13992a0ae89e51ddae7b3734e440a88ce7a7f

Reply:
> should we may be make the method private and deprecate it in 4.13?

I think so, yes. See https://github.com/contao/contao/pull/5155